internal/decision/engine.go

```diff
@@ -649,6 +650,31 @@ func (e *Engine) ReceiveFrom(from peer.ID, blks []blocks.Block) {
 	}
 	e.lock.RUnlock()
 
+	// If we found missing wants (e.g., because the peer disconnected, we have some races here)
+	// remove them from the list. Unfortunately, we still have to re-check because the user
+	// could have re-connected in the meantime.
+	if len(missingWants) > 0 {
+		e.lock.Lock()
+		for p, wl := range missingWants {
+			ledger, hasLedger := e.ledgerMap[p]
+			if hasLedger {
+				ledger.lk.RLock()
+			}
+			for _, k := range wl {
+				if hasLedger {
+					if _, has := ledger.WantListContains(k); has {
+						continue
+					}
+				}
+				e.peerLedger.CancelWant(p, k)
+			}
+			if hasLedger {
+				ledger.lk.RUnlock()
+			}
```

Comment:
nit: IMO it'd be easier to read this if we separated all the `hasLedger` pieces out and just had an if/else or an if + return.

Reply:
Yeah, good point.

Reply:
Done